### PR TITLE
Section popups

### LINKED ISSUE #802: Section popups
Show Section preview on interview page in a popup

- create new header to look good on popup, use goal popup as a reference
- put tasks just as-is and create a new task for fixing problems

### PR DESCRIPTION
**Pull request by Katrin-kudryash**
## PR includes

-   [x] Refactor
-   [x] Feature

## Related issues

Resolve #802

### CODE REVIEW

**Comment by Katrin-kudryash** [comment]
https://github.com/user-attachments/assets/df90cc81-3680-4fcf-929b-e52496edcc86

File: src/components/SectionPreview/SectionPreview.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                            <ModalContent className={s.ModalContent}>
+                                <Section section={section} visible={false} />
+                            </ModalContent>
+                        </div>
+                    </ModalPreview>
+                );
+            })}
+        </>
+    );
+};
```
Comment (asabotovich) [inline src/components/SectionPreview/SectionPreview.tsx:46]: I think we should have abstract component 
```
<Preview header={<HeaderComponentA />}>
    <ContentComponentA>
</Preview>
```

**Review by asabotovich — CHANGES_REQUESTED** [review]

File: src/components/SectionPreview/SectionPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+interface SectionProps {
+    sectionId: number;
+}
+
+export const SectionPreview = ({ sectionId }: SectionProps): JSX.Element => {
+    const { hidePreview } = usePreviewContext();
+    const sectionQuery = trpc.sections.getById.useQuery({ sectionId });
+
+    return (
+        <>
```
[the remark this answers is not in the document]
Comment (Katrin-kudryash) [reply src/components/SectionPreview/SectionPreview.tsx:21]: no
Untitled 2

File: src/contexts/previewContext.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+export const previewContext = createContext<PreviewContext>({
+    showSectionPreview: noop,
+    hidePreview: noop,
+});
+export const PreviewContextProvider: FC<PropsWithChildren> = ({ children }) => {
+    const [sectionId, setSectionId] = useState<number>();
+
+    const hidePreview = useCallback(() => {
+        setSectionId(undefined);
+    }, []);
```
[the remark this answers is not in the document]
Comment (Katrin-kudryash) [reply src/contexts/previewContext.tsx:21]: use 2 useMemo in PreviewContext components ?

File: src/components/HeaderPreview/HeaderPreview.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+                                    className={s.Avatar}
+                                    short
+                                    tooltip={section.interviewer.name}
+                                    size="s"
+                                    email={section.interviewer.email}
+                                    name={section.interviewer.name}
+                                />
+
+                                {section.interviewer.name}
+                            </Text>
```
Comment (asabotovich) [inline src/components/HeaderPreview/HeaderPreview.tsx:71]: Use component from harmony

File: src/components/HeaderPreview/HeaderPreview.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+interface HeaderPreviewProps {
+    pageTitle: string;
+    section: SectionWithRelationsAndResults;
+    href?: string;
+    date: Date;
+    readOnly?: boolean;
+}
+
+export const HeaderPreview: FC<HeaderPreviewProps> = ({ pageTitle, href, section, date, readOnly }) => {
```
Comment (asabotovich) [inline src/components/HeaderPreview/HeaderPreview.tsx:28]: Section Preview

File: src/components/Section/Section.tsx [hunk 6]
```diff
@@ -10,133 +9,89 @@ import { SectionWithRelationsAndResults } from '../../modules/sectionTypes';
...
 import { SectionFeedback } from '../SectionFeedback/SectionFeedback';
-import { SectionCancelationConfirmation } from '../SectionCancelationConfirmation/SectionCancelationConfirmation';
 import { Link } from '../Link';
 
 import { tr } from './Section.i18n';
 import s from './Section.module.css';
 
 interface SectionProps {
     section: SectionWithRelationsAndResults;
+    visible: boolean;
```
Comment (asabotovich) [inline src/components/Section/Section.tsx:24]: title: ReactNode

File: src/components/Previews/Previews.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    useEffect(() => {
+        hidePreview();
+    }, [pathname, hidePreview]);
+
+    return (
+        <>
+            {nullable(sectionId, (s) => (
+                <SectionPreview sectionId={s} />
+            ))}
+        </>
```
Comment (asabotovich) [inline src/components/Previews/Previews.tsx:21]: ```return nullable(sectionId, (s) => <SectionPreview sectionId={s} />);```

**Comment by Katrin-kudryash** [comment]
https://github.com/user-attachments/assets/d2157d6f-7de9-4ef9-9286-230247fa9833


https://github.com/user-attachments/assets/d88e2f27-7255-4325-96c1-3a37807931ec

**Review by asabotovich — APPROVED** [review]

File: src/components/ProblemList/ProblemList.tsx [hunk 8]
```diff
@@ -16,14 +16,16 @@ interface ProblemListProps {
     className?: string;
     style?: CSSProperties;
     isSmallSize?: boolean;
-    interviewId?: number;
+    interviewId: number;
```
Comment (tsumo) [inline src/components/ProblemList/ProblemList.tsx:19]: Problem list is used at the /problems where you don't have any interviews or sections

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/components/Problems/Problems.tsx [hunk 9]
```diff
@@ -1,13 +1,29 @@
...
-    </LayoutMain>
-);
+export const Problems = () => {
+    const router = useRouter();
+
+    const interviewId = parseNumber(router.query.interviewId);
+    const sectionId = parseNumber(router.query.sectionId);
+
+    return (
+        <LayoutMain pageTitle={tr('Problems')} aboveContainer={<ProblemFilterBar />}>
```
Comment (tsumo) [inline src/components/Problems/Problems.tsx:19]: Why filter bar is moved to aboveContainer?

File: src/components/SectionHeaderPreview/SectionHeaderPreview.i18n/ru.json [hunk 10]
```diff
@@ -0,0 +1,5 @@
+{
+    "Candidate": "",
```
Comment (tsumo) [inline src/components/SectionHeaderPreview/SectionHeaderPreview.i18n/ru.json:2]: translations

File: src/components/SectionPage/SectionPage.i18n/ru.json [hunk 11]
```diff
@@ -0,0 +1,7 @@
+{
+    "Section with {name} canceled": "Секция отменена по причине",
```
Comment (tsumo) [inline src/components/SectionPage/SectionPage.i18n/ru.json:2]: wrong translation

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/components/SectionHeaderPreview/SectionHeaderPreview.i18n/ru.json [hunk 12]
```diff
@@ -0,0 +1,5 @@
+{
+    "Candidate": "Кандидат",
+    "HR": "",
+    "Interviewer": "Интервьюер"
```
Comment (tsumo) [inline src/components/SectionHeaderPreview/SectionHeaderPreview.i18n/ru.json:4]: Собеседующий

File: src/components/Problems/Problems.tsx [hunk 13]
```diff
@@ -1,13 +1,25 @@
...
 
-export const Problems = () => (
-    <LayoutMain pageTitle={tr('Problems')} filterBar={<ProblemFilterBar title={tr('Problems')} />}>
-        <ProblemList />
-    </LayoutMain>
-);
+export const Problems = () => {
+    const router = useRouter();
+
+    const interviewId = parseNumber(router.query.interviewId);
```
Comment (tsumo) [inline src/components/Problems/Problems.tsx:15]: This page doesn't have interviewId and sectionId. Let's rename component to ProblemsPage to make it clear

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/components/ProblemsPage/ProblemsPage.tsx [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+import { ProblemList } from '../ProblemList/ProblemList';
+import { LayoutMain } from '../LayoutMain/LayoutMain';
+import { ProblemFilterBar } from '../ProblemFilterBar/ProblemFilterBar';
+
+import { tr } from './ProblemsPage.i18n';
+
+export const ProblemsPage = () => {
+    const router = useRouter();
+
+    const interviewId = parseNumber(router.query.interviewId);
```
Comment (tsumo) [inline src/components/ProblemsPage/ProblemsPage.tsx:15]: No interviewId and sectionId on this page

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/components/ProblemsPage/ProblemsPage.tsx [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+import { ProblemList } from '../ProblemList/ProblemList';
+import { LayoutMain } from '../LayoutMain/LayoutMain';
+import { ProblemFilterBar } from '../ProblemFilterBar/ProblemFilterBar';
+
+import { tr } from './ProblemsPage.i18n';
+
+export const ProblemsPage = () => {
+    const router = useRouter();
+
+    const sectionId = parseNumber(router.query.sectionId);
```
Comment (tsumo) [inline src/components/ProblemsPage/ProblemsPage.tsx:15]: This page never has sectionId in query

**Review by tsumo — CHANGES_REQUESTED** [review]

**Review by tsumo — APPROVED** [review]

**Pull request merged**